### ISSUE TITLE
Question: How to add user custom fields on plugin setup?

### ISSUE THREAD

**Issue by rica-carv** [e107inc/e107#1790]
I'm currently working on a plugin, and i'm wondering how can i add user custom fields when the plugin is setup, and how can that custom user fields be deleted when the plugin is uninstalled.....

Is that possible?

Currently the plugin is relying in a separate db table under the plugin name just to have a few user custom fields, but since v2.x added this user custom fileds functionality, i'm thinking on using it for this plugin.
Can it be donne?

**Comment by LaocheXe** [comment]
@rica-carv I've asked this before and was told to go into extended user fields -> Add Custom Field -> Type -> DB Table Field -> Select the table.

However I couldn't get this to work with my Plugin

**Comment by rica-carv** [comment]
> I've asked this before and was told to go into extended user fields -> Add Custom Field -> Type -> DB Table Field -> Select the table.

@LaocheXe  It's kind of ankward to install a plugin, and after have to create the fields the plugin needs "by hand" on the admin UI area....
My ideia is indeed use the custom fields in the core, introduced in V2.x, but is there anyway that the plugin can create that automatically?
If the base tool is already there, why not let a plugin use it, instead of "forcing" the plugin to create a separate db table with link to the user table?

**Comment by Moc** [comment]
Do you mean the "user extended fields"? Those can be defined in the plugin.xml

**Comment by LaocheXe** [comment]
@Moc can you give an example or tell us a plugin that has this?

**Comment by Moc** [comment]
The Forum plugin for example. 

http://e107.org/developer-manual/plugin-basics

**Comment by LaocheXe** [comment]
@Moc that is for forum post count correct? I think he wants one where the user can edit via profile settings. I know I would like have the function to do that.

**Comment by Moc** [comment]
Then e_user.php is probably what you are looking for: http://e107.org/developer-manual/plugin-basics/extending-core-functionality

**Comment by rica-carv** [comment]
Sorry, but somehow i can't open e107.org links..... Is site down??

**Comment by Moc** [comment]
Is a user allowed to edit those fields? 

I see what you mean now.. 
@CaMer0n Suggestions?

**Comment by rica-carv** [comment]
@Moc Yes, the user wil be allowed to edit those fileds, but since currently i guess there's no way to do it, i've already though of a workaround, ie, give the ability to edit them inside my own plugin....
Right now, my only big question is indded how to set those fileds inside the already established extended user fields, and use them, of course...

**Comment by CaMer0n** [comment]
@rica-carv I think you mean something like this?

	<extendedFields>
		<field name="status" text='My Status Field' type='EUF_TEXT' default='' active="true" values='true' system='false' read='e_UC_MEMBER' write='e_UC_MEMBER' />
	</extendedFields>

**Comment by rica-carv** [comment]
@CaMer0n Maybe, since i can't really figure out how that extended fields on the xml works.
Where are they stored? On the database? How can i fetch them from within PHP plugin code? How can a user define them on it's own logged in profile?

**Comment by CaMer0n** [comment]
@rica-carv This is from inside plugin.xml. During installation the extended fields are added to the database (user_extended and user_extended_struct). It is then up to each user to fill these fields in their usersettings. (which can be forced if required)

**Comment by lonalore** [comment]
I didn't want to open a new issue for my question, so I comment this...

**My issue:**
I want to create a custom user extended field... It would appear as a simple text field on the user settings form, except that it would use [Geocomplete jQuery plugin](https://github.com/ubilabs/geocomplete).

So I need to use a custom database table to store details for the selected place (Name, Latitude, Longitude... etc). I don't know how I can achive this... If I'm not wrong, I should use the [fields()](https://github.com/e107inc/e107/blob/master/e107_plugins/_blank/e_user.php#L29) method (in **e_user.php** addon file) combined with an [EUF_ADDON](https://github.com/e107inc/e107/blob/master/e107_plugins/_blank/plugin.xml#L28) field defined in **plugin.xml**. But I can not find documentation about the usage of **fields()** method. 

Any idea?

**Comment by lonalore** [comment]
It would be great if I can use something like in [e_admin.php](https://github.com/e107inc/e107/blob/master/e107_plugins/_blank/e_admin.php). I will be able to render my custom field widget, and with a process() method I will be able to process input data and store it in a custom table.

**Comment by lonalore** [comment]
Finally, I got it working (slightly differently)... I created a **EUF_TEXT** filed in my **plugin.xml**:
```xml
<extendedFields>
	<field name="location" type='EUF_TEXT' default="" active="true" system="false" text="..." read="e_UC_MEMBER" write="e_UC_MEMBER" applicable="e_UC_MEMBER" required="1" error="..."/>
</extendedFields>
```
Then I loaded a JS file on **usersettings.php**, which attaches Google Place Autocomplete behaviors to the texfield. This provides me a compatible location string for geocoding.

On the backend: I registered my callback function for "userdatachanged" event in my **e_event.php** addon file. In my callback, I use a PHP Geocoding API to get Latitude and Longitude, then I save location details in my custom database table for the user.
